@here/harp-mapview/lib/DataSource.ts

```diff
@@ -329,7 +325,10 @@ export abstract class DataSource extends THREE.EventDispatcher {
      * to a {@link MapView}.
      */
     get projection(): Projection {
-        return this.mapView.projection;
+        if (this.m_mapView === undefined) {
+            throw new Error("This DataSource was not added to MapView");
+        }
+        return (this.mapView as MapView).projection;
```

Comment:
```suggestion
        return this.m_mapView.projection;
```

Reply:
because you check for undefined above, you can use Typescripts smarts to avoid the cast

Reply:
right, nice.. will do that